modules/pulsepointLiteBidAdapter.js

```diff
@@ -1,301 +1,307 @@
  * bidder instead of separate requests per slot.
  */
-function PulsePointLiteAdapter() {
-  const bidUrl = window.location.protocol + '//bid.contextweb.com/header/ortb';
-  const ajaxOptions = {
-    method: 'POST',
-    withCredentials: true,
-    contentType: 'text/plain'
-  };
-  const NATIVE_DEFAULTS = {
-    TITLE_LEN: 100,
-    DESCR_LEN: 200,
-    SPONSORED_BY_LEN: 50,
-    IMG_MIN: 150,
-    ICON_MIN: 50,
-  };
+export const spec = {
 
-  /**
-   * Makes the call to PulsePoint endpoint and registers bids.
-   */
-  function _callBids(bidRequest) {
-    try {
-      // construct the openrtb bid request from slots
-      const request = {
-        imp: bidRequest.bids.map(slot => impression(slot)),
-        site: site(bidRequest),
-        device: device(),
-      };
-      ajax(bidUrl, (rawResponse) => {
-        bidResponseAvailable(bidRequest, rawResponse);
-      }, JSON.stringify(request), ajaxOptions);
-    } catch (e) {
-      // register passback on any exceptions while attempting to fetch response.
-      logError('pulsepoint.requestBid', 'ERROR', e);
-      bidResponseAvailable(bidRequest);
-    }
-  }
+  code: 'pulseLite',
 
-  /**
-   * Callback for bids, after the call to PulsePoint completes.
-   */
-  function bidResponseAvailable(bidRequest, rawResponse) {
-    const idToSlotMap = {};
-    const idToBidMap = {};
-    // extract the request bids and the response bids, keyed by impr-id
-    bidRequest.bids.forEach((slot) => {
-      idToSlotMap[slot.bidId] = slot;
-    });
-    const bidResponse = parse(rawResponse);
-    if (bidResponse) {
-      bidResponse.seatbid.forEach(seatBid => seatBid.bid.forEach((bid) => {
-        idToBidMap[bid.impid] = bid;
-      }));
-    }
-    // register the responses
-    Object.keys(idToSlotMap).forEach((id) => {
-      if (idToBidMap[id]) {
-        const size = adSize(idToSlotMap[id]);
-        const bid = createBid(STATUS.GOOD, bidRequest);
-        bid.bidderCode = bidRequest.bidderCode;
-        bid.cpm = idToBidMap[id].price;
-        bid.adId = id;
-        if (isNative(idToSlotMap[id])) {
-          bid.native = nativeResponse(idToSlotMap[id], idToBidMap[id]);
-          bid.mediaType = 'native';
-        } else {
-          bid.ad = idToBidMap[id].adm;
-          bid.width = size[0];
-          bid.height = size[1];
-        }
-        addBidResponse(idToSlotMap[id].placementCode, bid);
-      } else {
-        const passback = createBid(STATUS.NO_BID, bidRequest);
-        passback.bidderCode = bidRequest.bidderCode;
-        passback.adId = id;
-        addBidResponse(idToSlotMap[id].placementCode, passback);
-      }
-    });
-  }
+  aliases: ['pulsepointLite'],
 
-  /**
-   * Produces an OpenRTBImpression from a slot config.
-   */
-  function impression(slot) {
+  supportedMediaTypes: ['native'],
+
+  isBidRequestValid: bid => (
+    !!(bid && bid.params && bid.params.cp && bid.params.ct)
+  ),
+
+  buildRequests: bidRequest => {
```

Comment:
Can you please confirm the first param of this method is the array of bid requests? It appears so based on the call made from [here](https://github.com/prebid/Prebid.js/blob/master/src/adapters/bidderFactory.js#L27). This contradicts with the information [here](http://prebid.org/dev-docs/bidder-adapter-1.html), where ```bidderRequest``` param is an object, with a property ```bids```, holding the bid requests.

Reply:
Thanks for pointing this out Anand. Opened a PR (https://github.com/prebid/prebid.github.io/pull/380) on prebid.org to correct the docs. @snapwich will submit a PR to clarify the comments in the code.

Note that we also opted to rename the first parameter because there're actually two params formerly called "bidRequests" and "bidderRequest". This was kinda confusing, so we renamed the first to "validBidRequests". The second param isn't documented on prebid.org as it's expected to be needed only in rare cases, but is documented in the code.

Reply:
Thanks for clarifying @bretg